analyzers/src/SonarAnalyzer.CSharp/Rules/UnusedPrivateMember.cs

```diff
@@ -137,8 +131,17 @@ public sealed class UnusedPrivateMember : SonarDiagnosticAnalyzer
                         });
                 });
 
-        private static IEnumerable<Diagnostic> GetDiagnosticsForUnusedPrivateMembers(CSharpSymbolUsageCollector usageCollector, ISet<ISymbol> removableSymbols, string accessibility,
-            BidirectionalDictionary<ISymbol, SyntaxNode> fieldLikeSymbols)
+        private static bool IsTopLevelProgram(ISymbol symbol) =>
+            // If the symbol is named `Program` and the containing namespace is the global one, we will assume that we are analyzing a top level file.
+            // We cannot tell for sure, but in this case we prefer to have FNs instead of FPs, considering that there is a low chance to have
+            // a program class in the global namespace.
+            symbol.Name == "Program"
```

Comment:
Maybe you can check additionally the `symbol.GetMembers()` collection for an `IMethodSymbol` with name `<Main>$`.

Reply:
I think it would be useful to add this method in `SonarAnalyzer.Helpers.ISymbolExtensions` to reuse for other rules.